File: src/main/java/org/example/math_geometry/Power.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        def helper(x, n):
            if x == 0: return 0
            if n == 0: return 1

            res = helper(x * x, n // 2)  # instead of x * x, we can also do res = res * res
            # before returning (but that is slower)
            return x * res if n % 2 else res  # handle odd power
        res = helper(x, abs(n))
        return res if n >= 0 else 1 / res  # handle negative power

File: src/main/java/org/example/math_geometry/Power.kt
package org.example.math_geometry

import kotlin.math.abs

fun myPow(x: Double, n: Int): Double {
    fun helper(x: Double, n: Int): Double {
        if (n == 0) return 1.0
        if (x == 0.0) return 0.0

        val res = helper(x * x, n / 2)
        return if (n % 2 != 0) res * x else res
    }

    val res = helper(x, abs(n))
    return if (n < 0) 1 / res else res
}

fun main() {
    println(myPow(x = 2.00000, n = 10))
}